src/Catel.MVVM/Catel.MVVM.Shared/Services/DispatcherService.cs

```diff
@@ -56,12 +57,56 @@ public DispatcherService()
         /// <para />
         /// Internally, this property uses the <see cref="DispatcherHelper"/>, but can be overriden if required.
         /// </summary>
-        public virtual Dispatcher CurrentDispatcher
+        protected virtual Dispatcher CurrentDispatcher
         {
             get { return DispatcherHelper.CurrentDispatcher; }
         }
 #endif
 
+#if NET
+        /// <summary>
+        /// Executes the specified delegate asynchronously with the specified arguments on the thread that the Dispatcher was created on.
+        /// </summary>
+        /// <param name="action">The action.</param>
+        /// <returns>The task representing the action.</returns>
+        public Task InvokeAsync(Action action)
+        {
+            var dispatcher = CurrentDispatcher;
+
+#if NET40
+            return DispatcherExtensions.InvokeAsync(dispatcher, action);
+#else
+            return dispatcher.InvokeAsync(action).Task;
```

Comment:
I haven't test it. Maybe 2 separate InvokeAsync methods should be added (one for .Net 4.0 one for 4.5+)